### ISSUE TITLE
lsp-java-server-install-dir variable does not seem to work

### ISSUE THREAD

**Issue by ircurry** [emacs-lsp/lsp-java#470]
**Describe the bug**
No matter what I set  `lsp-java-server-install-dir` to, emacs always prompts me to install the language server. 

**To Reproduce**
Using the version of emacs, lsp, lsp-java, and jdt-language-server from my [flake](https://github.com/ircurry/cfg) and this elisp code:
```emacs-lisp
;; ===Java Tree-Sitter Mode===
(use-package java-ts-mode
  :mode "\\.java\\'")

;; ===lsp-java===
(use-package lsp-java
  ;:ensure t
  :after (lsp-mode cc-mode)
  :init
  :hook
  (envrc-mode . (lambda ()
		  (when (equal major-mode 'java-ts-mode)
		    (setq lsp-java-server-install-dir (concat (getenv "JDTLS_PATH") "/share/java/")))))
  (java-ts-mode . lsp-deferred)
  :config
  (defun lsp-java--ls-command ()
    (let ((jdtls-path (getenv "JDTLS_PATH"))
	  (jdtls-exec-options (list
			       "-configuration"
			       (concat (getenv "HOME") "/.jdtls/config_linux")
			       "-data"
			       (concat (getenv "HOME") "/.jdtls/java-workspace"))))
      (message (concat jdtls-path "/share/java/"))
      (append (list (concat jdtls-path "/bin/jdtls")) jdtls-exec-options))))
```

**Expected behavior**
Based on what I have gotten from the doc strings, [this issue](https://github.com/purcell/envrc/issues/55), and #421, I would expect for the server to execute as if that same jdtls version was installed at the default location.

**Screenshots**
![image](https://github.com/emacs-lsp/lsp-java/assets/114774855/4263dece-44bf-41f1-af96-d120dca801f6)

**Comment by ircurry** [comment]
Would this issue be fixed with #463 or am I misunderstanding what this pull request does?

**Comment by ircurry** [comment]
I have figured out the issue. The string concatenated should have been `"/share/java/jdtls/"` not `"/share/java/"`. A quick look at where `lsp-java-server-install-dir` is used, specifically `lsp-java--locate-server-jar`, and the files in the store would quickly revealed this and saved me so much time and pain.
